webapp/src/app/components/load-map/virtualMapNode.model.ts

```diff
@@ -17,22 +17,33 @@ export class VirtualMapNode {
         }
     }
 
-    public get name(): string {
-        return this.original.name;
+    public get names(): string[] {
+        const children = this.realChildren;
+        if (!children
+            || children.length !== 1 
+            || !children[0].isDirectory
+            || this.isRoot
+        ) {
+            return [this.original.name];
+        }
+
+        return [this.original.name].concat(...children[0].names);
     }
 
     public get path(): string | undefined {
         return this.original.path;
     }
 
     public get children(): VirtualMapNode[] | undefined {
-        if (!this.original.children) {
-            return undefined;
+        const result = this.realChildren;
+        if (result 
+            && result.length === 1 
+            && result[0].isDirectory 
+            && !this.isRoot
+        ) {
```

Comment:
if condition should be refactored into it's own method. It is the same as above, just negated

Reply:
Fixed in 9a3a38d